modules/fundamental/src/purl/service/mod.rs

```diff
@@ -230,41 +231,84 @@ impl PurlService {
         }
     }
 
-    pub async fn purl_by_purl<C: ConnectionTrait>(
+    #[instrument(skip(self, connection), err(level=tracing::Level::INFO))]
+    pub async fn fetch_purl_details<C: ConnectionTrait, I: AsRef<str> + Debug>(
```

Comment:
Accepting an `AsRef<str>` makes callers lives simpler as far as needing a `String`, a `&String` or a `&str`. They all implement `AsRef<str>` so they can all be passed. `Debug` is only there because of the `instrument` macro above.